Improve gas price management

fvictorio opened the pull request:
Closes #6.

This PR improves the way the gas price is obtained:

- Every some seconds (configurable), the gas price is obtained. This is done by fetching the oracle, calling the contract method or using the fallback price.
- This value is saved in memory and, when asked, this cached value is returned.

There are several changes that I made to be able to test stuff more easily:

- Added unit tests. Since we already had a `tests` directory, I renamed it to `scripts` and created a `test` directory for the unit tests.
- These scripts were getting the numnber of deposits/withdrawals to create from the environment. I think it's easier to pass this as an argument to the command, so I modified them to do this.

Finally, I modified the Travis configuration to run the linter and the unit tests before running the e2e tests.

fvictorio commented:
I just noticed that I may have done something wrong: when the oracle and the contract fail, the cached value is updated with the fallback value, but maybe the fallback value should be used just as the initial value (so that is used if the oracle and the contracts fail every time).

Another thing I don't like is that the fallback value will be used until the `setInterval` is executed the first time. So if this function is executed every 1 minute, during the first minute the fallback gas price will be used.

Both things are easy to fix, so let me know if you think they should be changed.

patitonar commented on src/services/gasPrice.js:
```diff
@@ -1,19 +1,99 @@
+let homeGasPriceInterval = null
+let foreignGasPriceInterval = null
+
+async function start() {
```
I think the sender needs the gas price for only one chain, Home or Foreign.

fvictorio commented:
Fixed @patitonar's comment and also modified the function that fetches the gas price so that it returns null if both the oracle and the contract fail. If that function returns null, the last cached gas price is used. This value is initialized with the fallback value, so the resulting behavior is:

- Start with the fallback value
- Fetch the gas price (success): the gas price is updated
- Fetch the gas price (fail): the last fetched value is used.
